activerecord/test/cases/relation/merging_test.rb

```diff
@@ -205,12 +205,22 @@ def test_merge_doesnt_duplicate_same_clauses
 
     only_david = Author.where("#{author_id} IN (?)", david)
 
-    assert_sql(/WHERE \(#{Regexp.escape(author_id)} IN \(1\)\)\z/) do
-      assert_equal [david], only_david.merge(only_david)
-    end
+    if ActiveRecord::Base.connection.adapter_name.match?(/mysql/i)
```

Comment:
Why do not use `if current_adapter?(:Mysql2Adapter)`?

Reply:
Because I knew a helper like that existed, but I somehow couldn't put my finger on it.

Thank you I'll cleanup my PR.

Any opinion on the rest?